holoviews/operation/datashader.py

```diff
@@ -89,6 +90,17 @@ class ResamplingOperation(Operation):
         Disable when you do not want the resulting plot to be interactive,
         e.g. when trying to display an interactive plot a second time.""")
 
+    precompute = param.Boolean(default=False, doc="""
+       Whether to apply precomputing operations, if available.
```

Comment:
Why 'if available'?

Reply:
When do you *not* want to precompute? I assume the default is False because precomputation has disadvantages?

Reply:
Sure, caching always has disadvantages, namely increased memory usage.

Reply:
Ok, then the docstring should mention that.

Reply:
>Why 'if available'?

regridding doesn't have or need precomputing.

Reply:
I'm also confused, this is not the sort of caching I was thinking of. I thought you cache from the previous call and if it is the same element, the data is available. If the data changes, then the cache is invalid and cleared.

Maybe you mean the operation *holds* onto memory between calls but from what I can tell, the transient memory usage when the operation is being applied should be the same.

Reply:
>I'm also confused, this is not the sort of caching I was thinking of. I thought you cache from the previous call and if it is the same element, the data is available. If the data changes, then the cache is invalid and cleared.

That's correct.

>Maybe you mean the operation holds onto memory between calls but from what I can tell, the transient memory usage when the operation is being applied should be the same.

Correct.

Reply:
I think the user doesn't need to know about regridding needing/not needing precomputation - whatever setting there is can be safely ignores so I wouldn't mention it, just to simplify the docstring and minimize confusion.

Reply:
Sounds good.

Reply:
Maybe something like:

> Whether to apply precomputing operations. Precomputing can speed up resampling operations by avoiding recomputation when the supplied element does not change at the cost of holding onto the memory used for this internal state between calls.